### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by Wils-FOURR-speed**
### Description
The following tasks were implemented in this part of the project:

- Created a class `Classroom.`
- Created the `has-many`/`belongs-to` relationship between `Classroom` and` Student.`
- Created a class` Book.`
- Created a class` Rental.`
- Created the` many-to-many` (also `has-many-through`) relationship between `Person `and` Book` using the intermediate class `Rental.`

### CODE REVIEW

File: student_class.rb [hunk 1]
```diff
@@ -9,4 +9,10 @@ def initialize(id, age, classroom, parent_permission: true, name: 'Unknown')
   def play_hooky
     '¯\\(ツ)/¯'
   end
+
+  def classroom=(classroom)
+    @classroom&.students&.delete(self)
+    @classroom = classroom
+    classroom.students << self unless classroom.nil?
+  end
```
Comment (thecodechaser) [inline student_class.rb:19]: - [ ] Please make sure the `classroom` variable is readable in this class for the associations to work as expected.
  - You can consider using this. `attr_reader :classroom`.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Book
+  attr_accessor :title, :rentals
+
+  def initialize(title)
+    @title = title
+    @rentals = []
+  end
+
```
Comment (thecodechaser) [inline book.rb]: - [ ] Please make sure the `@author` instance variable is initialized in the constructor same as `@title` as requested in the project requirements. Please also add `author` in the `attr_accessor` so it's readable and writable.
  - You can consider updating the constructor with the below code.
```ruby
def initialize(title, author)
    @title = title
    @author = author
    @rentals = []
  end

File: person_class.rb [hunk 3]
```diff
@@ -1,18 +1,40 @@
...
-  attr_reader :id
+require_relative 'nameable'
 
-  def initialize(_id, age, parent_permission: true, name: 'Unknown')
-    @id = Random.rand(1..1000)
-    @name = name
+class Person < Nameable
+  attr_accessor :age
+  attr_reader :id, :rentals, :classroom
+
```
Comment (thecodechaser) [inline person_class.rb]: - [ ] Please make sure the `rentals` variable is also writable for the associations to work as expected, you should add that in the `attr_accessor`.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Wils-FOURR-speed👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with associations. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @Wils-FOURR-speed, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**